File: test_data/cpp/test_main/aas_core_meta.v3/input/snippets/verification/reference_key_values_equal.cpp
bool ReferenceKeyValuesEqual(
  const std::shared_ptr<types::IReference>& that,
  const std::shared_ptr<types::IReference>& other
) {
  if (that->keys().size() != other->keys().size()) {
    return false;
  }

  const std::vector<
    std::shared_ptr<types::IKey>
  >& that_keys = that->keys();

  const std::vector<
    std::shared_ptr<types::IKey>
  >& other_keys = other->keys();

  for (size_t i = 0; i < that_keys.size(); ++i) {
    if (that_keys[i]->value() != other_keys[i]->value()) {
      return false;
    }
  }

  return true;
}

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/reference_key_values_equal.java
/**
* Check that the two references,'that' and
* 'other', are equal by comparing
* their {@link IReference#getKeys() keys} by
* {@link IKey#getValue() value}'s.
*/
public static boolean referenceKeyValuesEqual(
  IReference that,
  IReference other) {
  if (that.getKeys().size() != other.getKeys().size()) {
    return false;
  }

  for (int i = 0; i < that.getKeys().size(); i++) {
    if (!Objects.equals(that.getKeys().get(i).getValue(), other.getKeys().get(i).getValue())) {
      return false;
    }
  }

  return true;
}
